### PR TITLE
Use SOF to Query data from Single Directory

### PR DESCRIPTION
**Pull request by rnagweka**
You will learn
 -	How to define the database objects required to query structured data files.
 -	How to add a data source for a single data file and query the output of that file.

### CODE REVIEW

**Comment by MichaelCzcz** [comment]
@rnagweka Please resolve conflicts. Thank you.

**Comment by MichaelCzcz** [comment]
@rnagweka Are you still interested in this PR getting merged?

**Comment by rnagweka** [comment]
HI Michael,

Yes I want this pull request to get merged.

Regards,
Ronit Nagwekar

**Comment by MichaelCzcz** [comment]
@rnagweka Well, then please resolve the conflicts.

**Comment by MichaelCzcz** [comment]
@rnagweka There is only one conflict that needs to be resolved:
<<<<<<< master
 - A licensed SAP HANA Data Lake instance (non-trial / non-free tier).
=======
 - A licensed SAP HANA data lake instance (non-trial / non-free tier).
>>>>>>> master

File: tutorials/data-lake-file-containers-hdlfscli/data-lake-file-containers-hdlfscli.md [hunk 1]
```diff
@@ -128,14 +132,14 @@ Verify the certificate was signed by the CA.
 
 To obtain the subject string of a certificate in the RFC2253 format used in HDL Files authorizations (omit the "subject="prefix).
 
-**Note:** you will need this later when you configure authentication for HDL Files.
+**Note:** You will need this later when you configure authentication for HDL Files.
```
Comment (MichaelCzcz) [inline tutorials/data-lake-file-containers-hdlfscli/data-lake-file-containers-hdlfscli.md:135]: Please don't start a note with the label "Note". The formatting already makes it a note. 
Please check the [style guide](https://jam4.sapjam.com/wiki/show/MozDWaPczznL8NKeGEk0y9#mcetoc_1dtqfqchq21) for details.

File: tutorials/data-lake-file-containers-hdlfscli/data-lake-file-containers-hdlfscli.md [hunk 2]
```diff
@@ -173,29 +179,39 @@ Now click save at the top of the page.
...
+The following command wont return anything in the output.
 
-`hdlfscli -cert <PATH>/client.crt -key <PATH>/client.key -cacert <PATH>/ca.crt -k -s https://<REST API Endpoint> -filecontainer <Instance ID> ls`
+`hdlfscli -cert <PATH>\client.crt -key <PATH>\client.key -cacert <PATH>\ca.crt -k -s https://<REST API Endpoint> -filecontainer <Instance ID> ls`
 
-**[Optional]**: Configure a configuration file to make using the CLI simpler.
+**[Optional]:** Configure a configuration file to make using the CLI simpler.
 
-**Note**: The configuration will be placed in the user's root directory. It is saved as a JSON file that can be modified in any text editor.
+**Note:** The configuration will be placed in the **user's root directory**. It is saved as a JSON file that can be modified in any text editor.
```
Comment (MichaelCzcz) [inline tutorials/data-lake-file-containers-hdlfscli/data-lake-file-containers-hdlfscli.md:189]: Please don't start a note with the label "Note". The formatting already makes it a note.
Please check the [style guide](https://jam4.sapjam.com/wiki/show/MozDWaPczznL8NKeGEk0y9#mcetoc_1dtqfqchq21) for details.

File: tutorials/data-lake-hdlfs-dbx-connection/data-lake-hdlfs-dbx-connection.md [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+Once you click on OK. A **Data Lake File Store/Container** connection would be visible in the Databases section on the left in the Database Explorer. See Below.
+
+![Data Lake File Container](image-6.png)
+
+[DONE]
+[ACCORDION-END]
+
+[ACCORDION-BEGIN [Step 4: ](Verify the File container connection)]
+
+We need to verify whether we have added the right connection of File Container in DBX. We can do that by adding/uploading a file to the File container and checking it in the DBX.
```
Comment (MichaelCzcz) [inline tutorials/data-lake-hdlfs-dbx-connection/data-lake-hdlfs-dbx-connection.md:83]: Please don't use the "royal we" when addressing the user. Use "you" instead.
Please check the [style guide](https://jam4.sapjam.com/wiki/show/MozDWaPczznL8NKeGEk0y9#mcetoc_1dtvf24ta2) for details.

File: tutorials/data-lake-sof-multiple-directories/data-lake-sof-multiple-directories.md [hunk 4]
```diff
@@ -0,0 +1,238 @@
...
+- How to add a data source for a folder and subfolder directory and query the output of that file.
+
+
+## Intro
+
+Querying structured data files (CSV, ORC, Parquet) in a HANA Data Lake file container can be done using SQL on Files. Below you will find all of the steps required to start using SQL on Files.
+If you have not yet provisioned an SAP HANA data lake, here is a great tutorial on how to do so!
+Please go through the entire tutorial on **Use SQL on Files to query data from a single file/directory**
+
+We will be using the Orders table as a reference.
```
Comment (MichaelCzcz) [inline tutorials/data-lake-sof-multiple-directories/data-lake-sof-multiple-directories.md:31]: Please don't use the "royal we" when addressing the user. Use "you" instead.
Please check the [style guide](https://jam4.sapjam.com/wiki/show/MozDWaPczznL8NKeGEk0y9#mcetoc_1dtvf24ta2) for details.

File: tutorials/data-lake-sof-multiple-directories/data-lake-sof-multiple-directories.md [hunk 5]
```diff
@@ -0,0 +1,238 @@
...
+
+---
+
+### STEP 1: Create a SQL on Files remote table in Files service
+
+Note that the table definition will need to match the expected schema/data types of the columns in the structured file you are trying to query. Login to the SQL console as User1 or HDLADMIN and execute all the queries associated with this tutorial. I will be executing the queries through HDLADMIN.
```
Comment (MichaelCzcz) [inline tutorials/data-lake-sof-multiple-directories/data-lake-sof-multiple-directories.md:37]: Please fix grammar (the login, to log in). Who is "I"?

File: tutorials/data-lake-sof-multiple-directories/data-lake-sof-multiple-directories.md [hunk 6]
```diff
@@ -0,0 +1,238 @@
...
+
+Firstly, we create a SCHEMA in files service.
```
Comment (MichaelCzcz) [inline tutorials/data-lake-sof-multiple-directories/data-lake-sof-multiple-directories.md:39]: Please don't use the "royal we" when addressing the user. Use "you" instead.
Please check the [style guide](https://jam4.sapjam.com/wiki/show/MozDWaPczznL8NKeGEk0y9#mcetoc_1dtvf24ta2) for details.

File: tutorials/data-lake-sof-multiple-directories/data-lake-sof-multiple-directories.md [hunk 7]
```diff
@@ -0,0 +1,238 @@
...
+O_SHIPPRIORITY       integer,
+O_COMMENT             varchar(79)   
+) IN FILES_SERVICE;
+```
+<!-- border --> ![DBX Screenshot](image-1.png)
+
+For information about the parameter definitions and supported data types, see  [CREATE (Remote) TABLE Statement for Data Lake Relational Engine (HANA DB-Managed)](help.sap.com/docs/SAP_HANA_DATA_LAKE/a898e08b84f21015969fa437e89860c8/24e694b566814ad285cb32fe3e5d3928.html?state=DRAFT&version=2022_1_QRC)
+
+
+Next, we create a virtual table. Notice in the SQL below where the remote servers name goes and where the reference to the table in the Files Service goes. Over here, we will be creating an ORDERS VIRTUAL TABLE  in HDLRE that points to the ORDERS table that we just created in SQL On Files service.
```
Comment (MichaelCzcz) [inline tutorials/data-lake-sof-multiple-directories/data-lake-sof-multiple-directories.md:68]: Please don't use the "royal we" when addressing the user. Use "you" instead.
Please check the [style guide](https://jam4.sapjam.com/wiki/show/MozDWaPczznL8NKeGEk0y9#mcetoc_1dtvf24ta2) for details.

File: tutorials/data-lake-sof-multiple-directories/data-lake-sof-multiple-directories.md [hunk 8]
```diff
@@ -0,0 +1,238 @@
...
+
+### STEP 2: Optimize SQL on Files Performance Using Folder Hierarchies
+
+Another way to declare a data source with SQL on Files is using folder hierarchies. Folder hierarchies allow you to skip (or prune) files that do not need to be parsed in a query. Take a look at the following example.
+
+<!-- border --> ![Folder hierarchies](image-3.png)
+
+
+This diagram represents the folder structure hierarchy of the ORDERS table, for years 2020 to 2022 for ORDERS table there will be yearly data and from 2023 onwards to 2025, the table will have monthly data as well.
+In this case, we will create the folder structure in such a way that for ORDERYEAR=2020, it will query the data for that particular year i.e. 2020. We can also query the data for a particular month from that year and that is when ORDERMONTH and ORDERYEAR parameters comes into role. `SELECT * FROM ORDERS WHERE ORDERYEAR=2023 AND ORDERMONTH=05;` to skip parsing any data files under all the other directories for better performance.
```
Comment (MichaelCzcz) [inline tutorials/data-lake-sof-multiple-directories/data-lake-sof-multiple-directories.md:105]: Please don't use the "royal we" when addressing the user. Use "you" instead.
Please check the [style guide](https://jam4.sapjam.com/wiki/show/MozDWaPczznL8NKeGEk0y9#mcetoc_1dtvf24ta2) for details.

**Review by MichaelCzcz — CHANGES_REQUESTED** [review]

**Comment by jmmargo** [comment]
There is a conflict with the email address used to open this pull request.  Ronit will try to push these again.

**Pull request closed without merging**